src/com/palmergames/bukkit/towny/utils/SpawnUtil.java

```diff
@@ -0,0 +1,366 @@
+        	payee = town;
+        	break;        
+        case TOWN:
+			// Taking whichever is smaller, the cost of the spawn price set by the town, or the cost set in the config (which is the maximum a town can set their spawncost to.)
+			travelCost = Math.min(townSpawnPermission.getCost(town), townSpawnPermission.getCost());
+        	spawnPermission = String.format(spawnType.getTypeName() + " (%s)", townSpawnPermission);
+        	payee = town;
+        	break;        	
+        case NATION:
+            // Taking whichever is smaller, the cost of the spawn price set by the nation, or the cost set in the config (which is the maximum a nation can set their spawncost to.)
+            travelCost = Math.min(nationSpawnPermission.getCost(nation),nationSpawnPermission.getCost());
+        	spawnPermission = String.format(spawnType.getTypeName() + " (%s)", nationSpawnPermission);
+        	payee = nation;
+        	break;
+        default:
+        	break;
+        }
+
+		// Check if need/can pay.
+		try {
+			if ((!townyUniverse.getPermissionSource().has(player, PermissionNodes.TOWNY_COMMAND_TOWNYADMIN_TOWN_SPAWN_FREECHARGE.getNode())) && 
+					(travelCost > 0 && TownySettings.isUsingEconomy() && (resident.getHoldingBalance() < travelCost)))
+				throw new TownyException(notAffordMSG);
+		} catch (EconomyException ignored) {
+		}
+
+		// Used later to make sure the chunk we teleport to is loaded.
+		Chunk chunk = spawnLoc.getChunk();
+		
+		// Essentials tests.
+		boolean usingESS = plugin.isEssentials();
+
+		if (usingESS && !isTownyAdmin) {
+			try {
+				User essentialsUser = plugin.getEssentials().getUser(player);
+
+				// This jail check is specifically for essentials jails, not towny ones.
+				if (!essentialsUser.isJailed()) {
+
+					Teleport teleport = essentialsUser.getTeleport();
+					if (!chunk.isLoaded())
+						chunk.load();
+					// Cause an essentials exception if in cooldown.
+					teleport.cooldown(true);
+					teleport.teleport(spawnLoc, null, TeleportCause.COMMAND);
+				}
+			} catch (Exception e) {
+				TownyMessaging.sendErrorMsg(player, "Error: " + e.getMessage());
+				return;
+			}
+		}
+
+		// Actual taking of monies here.
+		if (!townyUniverse.getPermissionSource().has(player, PermissionNodes.TOWNY_COMMAND_TOWNYADMIN_TOWN_SPAWN_FREECHARGE.getNode())) {
+			if (!TownySettings.isTownSpawnPaidToTown())					
+				payee = TownyEconomyObject.SERVER_ACCOUNT;
+			// Show message if we are using an Economy and are charging for spawn travel.
+			try {
+				if (travelCost > 0 && TownySettings.isUsingEconomy() && resident.payTo(travelCost, payee, spawnPermission)) {
+					TownyMessaging.sendMsg(player, String.format(TownySettings.getLangString("msg_cost_spawn"), TownyEconomyHandler.getFormattedBalance(travelCost)));
+				}
+			} catch (EconomyException ignored) {
+			}
+		}
+
+		// If an Admin or Essentials teleport isn't being used, use our own.

```

Comment:
I would check if essentials is being used, for reasons mentioned above and also because essentials has a safe-teleport feature.

Reply:
Admins were always done separately to avoid any cooldown/warmup.